src/net/sourceforge/kolmafia/request/FightRequest.java

```diff
@@ -9468,6 +9471,9 @@ private static void payActionCost(final String responseText) {
 
       case SkillPool.METEOR_SHOWER:
         if (responseText.contains("consult your mental index") || skillSuccess) {
+          KoLConstants.activeEffects.remove(METEOR_SHOWERED);
+          KoLConstants.activeEffects.add(METEOR_SHOWERED);
+          KoLCharacter.recalculateAdjustments();
```

Comment:
I don't understand the mechanics.  Why are we removing then adding?  Why would recalculation not suffice?  Is this somehow defensive in case the effect had not "expired" after one turn and was then cast again?

Just curious, not blocking.

Reply:
Yeah this is to stop it being added twice. I could also do addToList I suppose. It needs to be added because it's not in the list yet (it only gets added to the list by the charpane refresh